edb/edgeql/codegen.py

```diff
@@ -548,13 +563,26 @@ def visit_Parameter(self, node: qlast.Parameter) -> None:
         self.write(param_to_str(node.name))
 
     def visit_StringConstant(self, node: qlast.StringConstant) -> None:
-        self.write(node.quote, node.value, node.quote)
+        if node.value.isprintable():
+            for d in ("'", '"', '$$'):
+                if d not in node.value:
+                    if '\\' in node.value and d != '$$':
+                        self.write('r', d, node.value, d)
+                    else:
+                        self.write(d, node.value, d)
+                    return
+        self.write(repr(node.value))
 
     def visit_RawStringConstant(self, node: qlast.RawStringConstant) -> None:
-        if node.quote.startswith('$'):
-            self.write(node.quote, node.value, node.quote)
-        else:
-            self.write('r', node.quote, node.value, node.quote)
+        if node.value.isprintable():
```

Comment:
I'm not sure about this.  Current code retains the `r` prefix.  This seems to treat both kinds of strings equally. Why?

Reply:
I don't see a reason why not? This is just the way the user writes strings. Semantically the strings are equivalent. It's like making a difference between single and double-quotes.